### ISSUE TITLE
 CA1416 Add support for constant/enum values passed as an argument to the guard methods. 

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn-analyzers#6299]
### Analyzer

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)

### Describe the improvement

Consider adding support for constant/enum values passed as an argument to the guard methods. 

Example:
```cs
private const int version = 8;

void M1()
{
    if (OperatingSystem.IsWindowsVersionAtLeast(version)) // not supported by the analyzer
    {
        Windows8Only(); // warns
    }

    if (OperatingSystem.IsWindowsVersionAtLeast(8)) // analyzer only support int literal
    {
        Windows8Only(); // not warn
    }
}

[SupportedOSPlatform("windows8.0")]
void Windows8Only() { }
```
This should work for custom APIs attributed with guard attributes too (SupportedOSPlatformGuard/UnsupportedOSPlatformGuard).

CC @AArnott @mavasani
